DisplayManager.cpp

```diff
@@ -0,0 +1,145 @@
+#include "DisplayManager.hpp"
+#include "SDL.h"
+
+using namespace std;
+
+#define SCALE_FACTOR 5
+
+#define DEFAULT_SCREEN_WIDTH 160
+#define DEFAULT_SCREEN_HEIGHT 144
+
+#define DEFAULT_SCREEN_PIXEL_COUNT DEFAULT_SCREEN_WIDTH *DEFAULT_SCREEN_HEIGHT
+
+#define SCREEN_WIDTH DEFAULT_SCREEN_WIDTH *SCALE_FACTOR
+#define SCREEN_HEIGHT DEFAULT_SCREEN_HEIGHT *SCALE_FACTOR
+
+#define WINDOW_NAME "Game"
+
+#define POSITION_TO_INDEX(x, y) ((y)*DEFAULT_SCREEN_WIDTH + (x))
+
+typedef struct
+{
+    SDL_Renderer *renderer;
+    SDL_Window *window;
+} App;
+
+App app;
+
+void initSDL(void)
+{
+    int rendererFlags, windowFlags;
+
+    rendererFlags = SDL_RENDERER_ACCELERATED;
+
+    windowFlags = 0;
+
+    if (SDL_Init(SDL_INIT_VIDEO) < 0)
+    {
+        printf("Couldn't initialize SDL: %s\n", SDL_GetError());
+        exit(1);
+    }
+
+    app.window = SDL_CreateWindow(WINDOW_NAME, SDL_WINDOWPOS_UNDEFINED, SDL_WINDOWPOS_UNDEFINED, SCREEN_WIDTH, SCREEN_HEIGHT, windowFlags);
+
+    if (!app.window)
+    {
+        printf("Failed to open %d x %d window: %s\n", SCREEN_WIDTH, SCREEN_HEIGHT, SDL_GetError());
+        exit(1);
+    }
+
+    SDL_SetHint(SDL_HINT_RENDER_SCALE_QUALITY, "linear");
+
+    app.renderer = SDL_CreateRenderer(app.window, -1, rendererFlags);
+
+    if (!app.renderer)
+    {
+        printf("Failed to create renderer: %s\n", SDL_GetError());
+        exit(1);
+    }
+}
+
+void renderScene(void)
+{
+    SDL_RenderPresent(app.renderer);
+}
+
+DisplayManager::DisplayManager()
+{
+
+    memset(&app, 0, sizeof(App));
+    _displayMatrix = new Color[DEFAULT_SCREEN_PIXEL_COUNT];
+    for (uint i = 0; i < DEFAULT_SCREEN_PIXEL_COUNT; i++)
+    {
+        _displayMatrix[i] = Color();
+    }
+    initSDL();
+}
+
+void DisplayManager::prepareScene()
+{
+    // performancewise : not the best, but most probably easier to adapt to our own game console.
```

Comment:
Since this file uses SDL directly you don't really have to worry about the game console, you can implement the more performant/easier to implement way and it should not have any effect on how the game console section will be coded later.